js/entities/Keyboard.js

```diff
@@ -14,9 +14,26 @@ const Keyboard = {
       return;
     }
 
-    Keyboard.left = e.which === vkLeft;
-    Keyboard.right = e.which === vkRight;
-    Keyboard.up = e.which === vkUp;
-    Keyboard.down = e.which === vkDown;
+    let lastKey = Object.keys(Keyboard)
+      .slice(0, -1)
+      .filter((e) => Keyboard[e])[0];
+
+    const leftCondition = e.which === vkLeft && lastKey !== "right";
+    const rightCondition = e.which === vkRight && lastKey !== "left";
+    const upCondition = e.which === vkUp && lastKey !== "down";
+    const downCondition = e.which === vkDown && lastKey !== "up";
+
+    if (
+      ![leftCondition, rightCondition, upCondition, downCondition].filter(
+        (boolean) => boolean === true
+      ).length > 0
```

Comment:
@ThiagoPereiraSje  aqui conferimos se as 4 condiçoes sao falsas . se for a funçao retorna , dessa maneira nao é assinalado nenhum novo valor as keys , fazendo entao que a cobinha nao mude de direçao!

Reply:
Entendi agora, mas acho que de toda a forma, pelo jeito que escrivi o jogo, só verificar a alteração na direção não vai fazer a cobrinha se mover ao contrário.

Pela forma que escrevi somente a cabeça da cobrinha é que muda de direção.

Acho que vai ser necessário criar uma lógica para fazer uma troca. Tipo: a cabeça ir para a posição final da calda e a ultima peça da calda ir para a posição onde está a cabeça.

Reply:
observa só como normalmente o jogo é:
https://www.google.com/search?q=snake+game&oq=snake+game&aqs=chrome..69i57j0i131i433i512j46i512j0i512l3j69i60l2.3752j0j7&sourceid=chrome&ie=UTF-8
 o que acontece é quando vc clica na direçao oposta ? apenas nada!

Reply:
Ah agora entendi! Isso ai dá pra corrigir fácil.
Eu achei que quando fosse a direção inversa a cobrinha tinha que andar ao contrário.
Achei que era isso que vc tava falando.

Reply:
Tava pensando aqui, nesse caso acho que só isso aqui resolveria:

const vkLeft = 37;
const vkRight = 39;
const vkUp = 38;
const vkDown = 40;

const Keyboard = {
  left: false,
  right: true,
  up: false,
  down: false,

  keydown: function (e) {
    if (![vkLeft, vkRight, vkDown, vkUp].includes(e.which)) {
      return;
    }

    const isLeft = e.which === vkLeft;
    const isRight = e.which === vkRight;
    const isUp = e.which === vkUp;
    const isDown = e.which === vkDown;

    if (isLeft && Keyboard.right) {
	return;
    }

    if (isRight && Keyboard.left) {
	return;
    }

    if (isUp && Keyboard.down) {
	return;
    }

    if (isDown && Keyboard.up) {
	return;
    }

    Keyboard.left = isLeft;
    Keyboard.right = isRight;
    Keyboard.up = e.which === isUp;
    Keyboard.down = e.which === isDown;
  },
};

Reply:
@ThiagoPereiraSje da o commit nessa pr e merge quando der ^^

Reply:
blz